selfdrive/ui/replay/filereader.h

```diff
@@ -1,70 +1,90 @@
 #pragma once
 
-#include <thread>
+#include <capnp/serialize.h>
 
 #include <QElapsedTimer>
 #include <QMultiMap>
 #include <QNetworkAccessManager>
-#include <QReadWriteLock>
 #include <QString>
-#include <QVector>
-#include <QWidget>
+#include <QThread>
+#include <unordered_map>
+#include <vector>
 
-#include <bzlib.h>
-#include <capnp/serialize.h>
-#include <kj/io.h>
 #include "cereal/gen/cpp/log.capnp.h"
 
-#include "tools/clib/channel.h"
-
 class FileReader : public QObject {
   Q_OBJECT
 
 public:
-  FileReader(const QString& file_);
-  void startRequest(const QUrl &url);
-  ~FileReader();
-  virtual void readyRead();
-  void httpFinished();
-  virtual void done() {};
-
-public slots:
-  void process();
+  FileReader(const QString &fn, QObject *parent = nullptr);
+  void read();
+  void abort();
 
-protected:
-  QNetworkReply *reply;
+signals:
+  void finished(const QByteArray &dat);
+  void failed(const QString &err);
 
 private:
-  QNetworkAccessManager *qnam;
-  QElapsedTimer timer;
-  QString file;
+  void startHttpRequest();
+  QNetworkReply *reply_ = nullptr;
+  QUrl url_;
 };
 
-typedef QMultiMap<uint64_t, cereal::Event::Reader> Events;
+enum CameraType {
```

Comment:
fine here for now, but we should really clean this up in camerad and have a nice header for all this